### PR TITLE
Add support for LocalizedError

### PR DESCRIPTION
**Pull request by ecamacho**
The package's `Error`s were not conforming to LocalizedError, making hard to get the error cause when used from another SwiftPackage. 

Also, the `Reporter` files were missing their Copyright header.

### CODE REVIEW

File: Sources/XCLogParser/loglocation/LogError.swift [hunk 1]
```diff
@@ -20,12 +20,42 @@
...
+    public var errorDescription: String? { return description }
+
+}
+
+extension LogError: CustomStringConvertible {
+
+    public var description: String {
+        switch self {
+        case .noDerivedDataFound:
+            return "We couldn't find a the derivedData directory. " +
```
Comment (BalestraPatrick) [inline Sources/XCLogParser/loglocation/LogError.swift]: small typo

File: Sources/XCLogParser/loglocation/LogError.swift [hunk 2]
```diff
@@ -20,12 +20,42 @@
...
+            "If you use a custom derivedData dir, use the --derivedData option to pass it. "
```
Comment (BalestraPatrick) [inline Sources/XCLogParser/loglocation/LogError.swift]: dir => directory

File: Sources/XCLogParser/loglocation/LogError.swift [hunk 3]
```diff
@@ -20,12 +20,42 @@
...
+                "If the log is in a custom derivedData dir, use the --derivedData option. " +
+            "You can also pass the full path to the xcactivity log with the --file option"
+        case .xcodeBuildError(let error):
+            return error
+        case .readingFile(let path):
+            return "Can't read file \(path)"
+        case .invalidFile(let path):
+            return "\(path) is not a valid xcactivitylog file"
+        case .noLogManifestFound(let path):
+            return "We couldn't find a logManiifest in the path \(path). " +
```
Comment (BalestraPatrick) [inline Sources/XCLogParser/loglocation/LogError.swift]: `logManifest`

**Comment by ecamacho** [comment]
@BalestraPatrick Thanks! I fixed the typos

**Pull request merged**